pkg/virt-launcher/virtwrap/converter/compute/hypervisor.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package compute
+
+import (
+	"fmt"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type HypervisorDomainConfigurator struct {
+	allowEmulation bool
+	kvmAvailable   bool
+}
+
+// NewHypervisorDomainConfigurator creates a new hypervisor domain configurator
+func NewHypervisorDomainConfigurator(allowEmulation bool, kvmAvailable bool) *HypervisorDomainConfigurator {
```

Comment:
No need to return a pointer:
```suggestion
func NewHypervisorDomainConfigurator(allowEmulation, kvmAvailable bool) HypervisorDomainConfigurator {
```